### PR TITLE
Fixes to the the merge with MAPS 8.1

### PR DESCRIPTION
**Pull request by tanyasmirnova**
The title above should be a 1 line short summary of the pull request (i.e. what the project the PR represents is intended to do).
Fixes to the the merge with MAPS 8.1
Enter a description of this PR. This should include why this PR was created, and what it does.
The 8.,1 version of MPAS has a new driver for sea ice which was using only Noah sea ice model. To fix the merge RUC ice model had to be added to the driver_seaice. 
Testing and relations to other Pull Requests should be added as subsequent comments.
The fixes are tested for the 24 feb 2023 case on jet: /mnt/lfs4/BMC/wrfruc/smirnova/MPAS_conus13_20230224_mpasGSL
See the below examples for more information.
https://github.com/MPAS-Dev/MPAS/pull/930
https://github.com/MPAS-Dev/MPAS/pull/931

### CODE REVIEW

File: src/core_atmosphere/physics/physics_wrf/module_sf_ruclsm.F [hunk 1]
```diff
@@ -1706,6 +1709,7 @@ subroutine sfctmp (spp_lsm,rstochcol,fieldcol_sf,             &
        endif
 
     if(seaice .lt. 0.5) then
+      if(doseaice) goto 119
```
Comment (AndersJensen-NOAA) [inline src/core_atmosphere/physics/physics_wrf/module_sf_ruclsm.F]: Can you get rid of this goto?

File: src/core_atmosphere/physics/physics_wrf/module_sf_ruclsm.F [hunk 2]
```diff
@@ -1756,13 +1760,14 @@ subroutine sfctmp (spp_lsm,rstochcol,fieldcol_sf,             &
...
-      else
+     else
 !-- change albedo when no fresh snow and snow albedo is higher than 0.5
         alb = min(albsn,max(albsn - 0.1*(soilt - 263.15)/       &
                 (273.15-263.15)*albsn, albsn - 0.05))
-      endif
+     endif
     else
 !----- snow on ice
+     if (.not. doseaice) goto 119
```
Comment (AndersJensen-NOAA) [inline src/core_atmosphere/physics/physics_wrf/module_sf_ruclsm.F]: Can you get rid of this goto?

Comment (AndersJensen-NOAA) [reply src/core_atmosphere/physics/physics_wrf/module_sf_ruclsm.F]: And others below.

**Review by AndersJensen-NOAA — CHANGES_REQUESTED** [review]

File: src/core_atmosphere/physics/physics_wrf/module_mp_thompson.F
Comment (joeolson42) [inline src/core_atmosphere/physics/physics_wrf/module_mp_thompson.F:1]: how did this get removed? Is it intended? I suppose it doesn't matter with Anders' submodule...

**Review by joeolson42 — APPROVED** [review]

File: src/core_atmosphere/physics/physics_wrf/module_mp_thompson.F
Comment (jderrico-noaa) [reply src/core_atmosphere/physics/physics_wrf/module_mp_thompson.F:1]: @AndersJensen-NOAA does this matter?

**Comment by AndersJensen-NOAA** [comment]
@jderrico-noaa @joeolson42 @tanyasmirnova This PR compiles and runs for a 3-km CONUS case (I just ran a short 3-h test). This can be merged. Note that we won't be using the deleted version of the microphysics, but it probably should be committed back so that our version of MPAS stays in line with the other versions out there.

**Comment by jderrico-noaa** [comment]
@tanyasmirnova can you restore the deleted microphysics?  After that I'll merge this!

**Comment by tanyasmirnova** [comment]
Janet,
I found a problem with soil moisture in this PR. I am looking at this
today, and hopefully will fix the problem towards the end of the day.
I can also restore the MP Thompson in physics_wrf.
Thanks,
Tanya

**Comment by tanyasmirnova** [comment]
Janet and All,
First, about module_mp_thompson.F - it gets deleted automatically when the
Thompson submodule is checked out.
I haven't deleted it myself.

Second, at this point I think I want to give up on PR#8. I'll try to
explain briefly.
I noticed that after the second call to RUC LSM from driver_seaice, soil
moisture on land got corrupted. I am not quite sure why, but maybe the
reason is that the names of variables for land and ice are the same (in
RRFS the names are *_land and *_ice). When we get from MPAS a variable from
the driver_seaice, it should update only sea-ice points, but it appears
that land points are also modified or corrupted.
When I removed soil moisture from updated variables in driver_seaice,
then it looked good.
However, there are several other variables that are updated for sea ice
points, and cannot be removed. And now I am not sure they are not corrupted
over land.
Also, in the MPAS 8.1, 2-m diagnostics is called out of driver_seaice which
is called after driver_lsm, and I have suspicions that over the land it
will not be correct.
Because we have limited time for figuring all this out, I think we should
go with a simple solution #2: keep one call to RUC LSM from driver_lsm, and
also keep RUC 2-m diagnostics in driver_lsm. In this case we do not need to
call dirver_seaice from mpas_atmphys_driver.F with RUC LSM, and should add
IF around driver_seaice call.
This solution has less changes to the code, but most likely will not be
accepted by Laura Fowler.

I am running a test now with Solution #2. I will see the results tomorrow
morning. If I see more reasonable differences between 8.1 and 8.0 code, I
will make a new PR.
Let me know what you think about this.
Thank you,
Tanya

**Comment by tanyasmirnova** [comment]
@jderrico-noaa @AndersJensen-NOAA 
Janet and All,
The results look much more reasonable now with the Solution #2. 
I am going to submit the PR.
Anders, could you please repeat your 3-km case. 
In my 15-km CONUS test the 2-m temperature differences compared to 8.0 code looks like:
Screenshot 2024-05-03 at 9 31 32 AM
And for 2-m Qv:
Screenshot 2024-05-03 at 9 32 23 AM
Top-level soil moisture looks like:
Screenshot 2024-05-03 at 9 38 46 AM
The difference look much more reasonable than those shown at the Wednesday's meeting.

**Comment by tanyasmirnova** [comment]
PR for solution 2 is submitted: https://github.com/NOAA-GSL/MPAS-Model/pull/9

**Pull request closed without merging**